lib/install/mutate-into-logical-tree.js

```diff
@@ -81,7 +81,12 @@ function translateTree_ (tree, seen) {
   pkg._dependencies = pkg.dependencies
   pkg.dependencies = {}
   tree.children.forEach(function (child) {
-    pkg.dependencies[moduleName(child)] = translateTree_(child, seen)
+    const dep = pkg.dependencies[moduleName(child)] = translateTree_(child, seen)
+    if (child.fakeChild) {
+      dep.missing = true
+      dep.optional = child.package._optional
+      dep.requiredBy = child.package._spec
```

Comment:
This is where we're producing `read-installed`'s output for `ls` to consume. In it's format `requiredBy` is the spec that required the missing dep.